mev-rs/src/blinded_block_provider/api/server.rs

```diff
@@ -49,13 +49,19 @@ async fn handle_open_bid<B: BlindedBlockProvider>(
     State(builder): State<B>,
     Json(block): Json<serde_json::Value>,
 ) -> Result<Json<VersionedValue<ExecutionPayload>>, Error> {
-    let maybe_capella_block = capella::SignedBlindedBeaconBlock::deserialize(&block);
-    let mut block = match maybe_capella_block {
-        Ok(block) => SignedBlindedBeaconBlock::Capella(block),
-        Err(err) => match bellatrix::SignedBlindedBeaconBlock::deserialize(block) {
-            Ok(block) => SignedBlindedBeaconBlock::Bellatrix(block),
-            Err(_) => return Err(ApiClientError::from(err).into()),
-        },
+    // TODO: Using the optional `Eth-Consensus-Version` header once clients have implemented it.
+    let mut block = if block["message"].is_null() {
+        deneb::SignedBlindedBlockAndBlobSidecars::deserialize(&block)
+            .map(SignedBlindedBeaconBlock::Deneb)
+            .map_err(ApiClientError::from)?
+    } else {
+        capella::SignedBlindedBeaconBlock::deserialize(&block)
+            .map(SignedBlindedBeaconBlock::Capella)
+            .or_else(|_| {
+                bellatrix::SignedBlindedBeaconBlock::deserialize(&block)
+                    .map(SignedBlindedBeaconBlock::Bellatrix)
+            })
+            .map_err(ApiClientError::from)?
```

Comment:
how reliable is this "message will be null" condition?

I would imagine we would just extend the "try to deser and move to a prior fork if it fails" pattern like was already here

open to this if it is reliable and also agree we should try to patch this upstream so its simpler to handle in implementations

Reply:
I am actually tempted to switch back the existing pattern, but wanted to see if you have any thoughts :) 

In terms of reliability i'm guessing it's probably the same as the other approach, because the deneb type has an extra layer and does not have the "message" field on the top layer. (I've tested this using [mock-relay](https://github.com/realbigsean/mock-relay) and it worked)

My original thought was - it would probably be more performant to check for a field than deserializing the entire JSON, but I haven't done any benchmarking and it was soley based on assumption - thinking about this again, it probably does the same thing under the hood, and I would say it's premature optimisation without doing any benchmarking. I think I'd probably revert to the existing approach as it's more readable, and consider this again if we run into performance issues.

ps. I vaguely remember reading an issue somewhere that you're looking to add polymorphism support to these fork versioned types, so didn't really put in too much time thinking about this tbh.

Reply:
I'd say we go w/ the existing pattern for now, as it is simple and easy to reason about ("if this can be interpreted as a Deneb message, lets do that; if not, let's try a capella message; if not ...")

Reply:
Cool make sense - I think the upside is once we hit the deneb fork, it should always hit the first branch, so we're actually better off without the extra field access.

Updated in https://github.com/ralexstokes/mev-rs/pull/124/commits/ac8b5808514060c38a6d169a87b3db87acd737c4